packages/callbacks/lib/tizen_interop_callbacks.dart

```diff
@@ -0,0 +1,281 @@
+
+/// Class for storing data about registered callback.
+///
+/// Obtained by calling [TizenInteropCallbacks.register()].
+/// Then [regCallbackPtr] and [regUserData] should be passed to native API
+/// callback setup.
+class RegisteredCallback<NativeCbType extends Function> {
+  /// Unique identifier of the callback instance.
+  final int id;
+
+  /// Name of the Native API callback.
+  final String nativeName;
+
+  /// Name of the proxy callback. [nativeName] prefixed with
+  /// `platform_blocking_` or `platform_non_blocking_`.
+  final String platformCbName;
+
+  /// Index identifying which of multiple proxies is used for this registered callback.
+  final int proxyId;
+
+  /// The pointer that will be used in native callback setter/adder.
+  final Pointer<NativeFunction<NativeCbType>> regCallbackPtr;
+
+  /// User data pointer to be used in native callback setter/adder.
+  final Pointer<Void> regUserData;
+
+  /// See [RegisteredCallback] class members documentation.
+  RegisteredCallback(this.id, this.nativeName, this.platformCbName,
+      this.proxyId, this.regCallbackPtr, this.regUserData);
+}
+
+/// Class which manages wrappers and registration of callbacks.
+///
+/// This class should be instantiated only once, on the main thread of
+/// the main isolate - the one with your main() entry point / runApp() call.
+///
+/// There purpose of this class is to solve the issue related to error message:
+/// `Cannot invoke native callback outside an isolate.` If you are not
+/// receiving such error message, that means you don't need this class.
+///
+/// Before your callback can be called by some Tizen Native API modules,
+/// you should register it and pass [RegisteredCallback.regCallbackPtr]
+/// and [RegisteredCallback.regUserData] you will receive to Native API:
+/// ``` dart
+/// var levelCb = callbacks.register<
+///       Void Function(Int32, Pointer<Void>, Pointer<Void>)
+///     >('device_changed_cb', Pointer.fromFunction(_batteryChanged));
+/// tizen.device_add_callback(2, levelCb.regCallbackPtr, levelCb.regUserData);
+/// ```
+/// See [register()] for additional information.
+class TizenInteropCallbacks {
+  final DynamicLibrary _process = DynamicLibrary.process();
+  late Pointer Function(int, Pointer, Pointer, Pointer, int)
+      _registerWrappedCallback;
+  late Pointer Function(int) _unregisterWrappedCallback;
+  late bool Function(Pointer<Utf8>) _platformProxyCallbackExists;
+  late void Function(Pointer) _runCallbackInNativeLayer;
+  static TizenInteropCallbacks? _instance;
+
+  int _curCallbackId = 0;
+  // ignore: non_constant_identifier_names
+  final int _PROXY_ID_COUNT = 5;
+  final _multiProxyIds = <String, List<bool>>{};
+
+  /// Initializes the [TizenInteropCallbacks].
+  ///
+  /// This class should be instantiated only once, on the main thread of
+  /// the main isolate - the one with your main() entry point / runApp() call.
+  TizenInteropCallbacks() {
+    if (_instance != null) {
```

Comment:
a proper singleton would be better, necessary even in case this is used by independent plugins

Reply:
changed to use singleton through factory constructor